### ISSUE TITLE
SSR will error because `esModuleInterop` and `allowSyntheticDefaultImports` changes import semantics? of dayjs in particular

### ISSUE THREAD

**Issue by nopeless** [nuxt/framework#4678]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.1`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `ssr`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-dsfksc?file=plugins%2Fdayjs.ts

### Describe the bug

```
Nuxt CLI v3.0.0-rc.1                                                                                              12:53:31
                                                                                                                  12:53:31
  > Local:    http://localhost:3000/ 
  > Network:  http://192.168.1.104:3000/

ℹ Vite client warmed up in 1239ms                                                                                 12:53:36
ℹ Vite server warmed up in 733ms                                                                                  12:53:36
✔ Vite server built in 1067ms                                                                                     12:53:37
✔ Nitro built in 237 ms                                                                                     nitro 12:53:37
[nitro] [dev] [unhandledRejection] TypeError: __vite_ssr_import_1__ is not a function
[nitro] [dev] [unhandledRejection] TypeError: __vite_ssr_import_1__ is not a function
```

This specifically happens because in dayjs docs you aren't allowed to import it via import * as dayjs

https://day.js.org/docs/en/installation/typescript

However, I think the error can be a bit more descriptive and perhaps give pointers

### Additional context

Sorry for the verbose title. I don't know how to put it in simpler terms

I think it might be worth adding a guard when loading the modules

**Comment by danielroe** [comment]
In your reproduction, it looks like everything is working fine with `import dayjs from 'dayjs'`. Could you clarify what you think the Nuxt issue/enhancement is?

**Comment by nopeless** [comment]
> In your reproduction, it looks like everything is working fine with `import dayjs from 'dayjs'`. Could you clarify what you think the Nuxt issue/enhancement is?

It works fine but I think nuxt should be able to detect these issues. An error explaining why a module failed to reload is better than what seems to be an internal bug
